### ISSUE TITLE
Support older autoscaling versions

### ISSUE THREAD

**Issue by thatsmydoing** [zalando-incubator/kube-metrics-adapter#55]
Currently, the adapter only tries to discover autoscaling/v2beta2 HPAs. This means it can only be used on newer kubernetes versions that provide autoscaling/v2beta2. It would be nice to have support for the old autoscaling/v2beta1 API as well.

Alternatively, the README could have a support matrix showing which versions of the adapter support which versions of kubernetes.

**Comment by agolomoodysaada** [comment]
Doesn't this mean that this project doesn't support all k8s versions?
We should have clear kubernetes version compatibility listed.
Also the shown example is using v2beta1 : https://github.com/zalando-incubator/kube-metrics-adapter/blob/master/example/deploy/hpa.yaml#L1

**Comment by mikkeloscar** [comment]
I think it would be reasonable to strive for supporting the latest 3 minor versions of Kubernetes which is the same support guarantees as provided for [Kubernetes itself](https://kubernetes.io/docs/setup/version-skew-policy/#supported-versions).

We currently don't do this since `autoscaling/v2beta2` was introduced in `v1.12`

**Comment by agolomoodysaada** [comment]
Weird... I tried k8s `1.11` and `1.12` but still not seeing the `v2beta2` version on GKE. For reference I'm trying on GKE `v1.12.7-gke.10`. on `v2beta1` shows up in `kubectl api-versions`. 

I think supporting `v2beta1` would support a lot more versions.

**Comment by mikkeloscar** [comment]
I have not tried to run it on GKE, but it's listed in the changelog: https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.12.md#sig-autoscaling and works on our v1.12 clusters, so I would expect it to also be there...

**Comment by agolomoodysaada** [comment]
I filed a GKE issue with them to see what's the issue. But in general, shouldn't this project support more autoscaling versions such as `v2beta1`?

**Comment by mikkeloscar** [comment]
> But in general, shouldn't this project support more autoscaling versions such as v2beta1?

This sounds reasonable yes, but we should also understand why this doesn't work in GKE when it should work with 1.12. Thanks for opening an issue!

Also note that currently you can still use `v2beta1` hpas in a cluster that also supports the `v2beta2` api, it only means the adapter can not run in clusters without `v2beta2` support. Which means anything before `v1.12`

**Comment by agolomoodysaada** [comment]
I will write the GKE team's response here. In the meantime, can we add a flag or something that tells the adapter to check for v2beta1 hpas instead of forcing v2beta2?

**Comment by agolomoodysaada** [comment]
The GKE team is working on adding support for v2beta2 at the moment with no ETA yet.

**Comment by mikkeloscar** [comment]
> The GKE team is working on adding support for v2beta2 at the moment with no ETA yet.

Thanks for looking into this. I'm at KubeCon and will also try to ask some of the GKE people about it.

**Comment by mikkeloscar** [comment]
GKE v1.13.5 also doesn't seem to have the API enabled:

```
$ kubectl api-versions
admissionregistration.k8s.io/v1beta1
apiextensions.k8s.io/v1beta1
apiregistration.k8s.io/v1
apiregistration.k8s.io/v1beta1
apps/v1
apps/v1beta1
apps/v1beta2
authentication.k8s.io/v1
authentication.k8s.io/v1beta1
authorization.k8s.io/v1
authorization.k8s.io/v1beta1
autoscaling/v1
autoscaling/v2beta1
batch/v1
batch/v1beta1
certificates.k8s.io/v1beta1
cloud.google.com/v1beta1
coordination.k8s.io/v1beta1
extensions/v1beta1
metrics.k8s.io/v1beta1
networking.gke.io/v1beta1
networking.k8s.io/v1
policy/v1beta1
rbac.authorization.k8s.io/v1
rbac.authorization.k8s.io/v1beta1
scalingpolicy.kope.io/v1alpha1
scheduling.k8s.io/v1beta1
storage.k8s.io/v1
storage.k8s.io/v1beta1
v1
```

Will try to find the right people today, so far no-one from GKE knew anything :(

**Comment by mikkeloscar** [comment]
So I talked with the autoscaling team from Google and they are aware of the issue and working on a fix, but it may take a bit to get rolled out. Maybe we can see about falling back to supporting `v2beta1` in the meantime.

**Comment by agolomoodysaada** [comment]
The issue is now officially submitted to google team and can be tracked here: https://issuetracker.google.com/issues/135624588

**Comment by mikkeloscar** [comment]
Pinged on the GKE issue today as this is still not enabled in the latest GKE version :(

**Comment by agolomoodysaada** [comment]
Would like to confirm that this is still not supported on `v1.13.7-gke.8`.
Thanks for starting efforts on #64 !

**Comment by mikkeloscar** [comment]
I tested `v1.13.7-gke.8` the other day, still not enabled (sorry forgot to mention the version).

Would you mind testing #64 in GKE if you have time, it should just work out of the box now. There is an image available built from the PR: `registry.opensource.zalan.do/teapot/kube-metrics-adapter-test:pr-64-3`

**Comment by mikkeloscar** [comment]
Issue should be fixed and is included in the first official release: https://github.com/zalando-incubator/kube-metrics-adapter/releases/tag/v0.0.1

**Comment by agolomoodysaada** [comment]
I would like to confirm that GKE works well now with `v2beta1.` Thank you so much for this change.

**Comment by mikkeloscar** [comment]
`autoscaling/v2beta2` still not enabled in GKE `1.14.6-gke.13` :man_shrugging:

**Comment by agolomoodysaada** [comment]
It makes no sense

**Comment by mikkeloscar** [comment]
I asked for help on twitter: https://twitter.com/mikkeloscar/status/1182545645032349696

After my tweet someone was at least assigned to the bug ticket.

**Comment by mikkeloscar** [comment]
Finally some progress on this:

> I've enabled the v2beta2 in the upcoming GKE 1.16 version (no timeline yet).  I also backported the change to 1.15.5 which will become available after 1.15 goes GA, hopefully soon.
> 
> My recommendation is to wait for 1.15.5 to become available and upgrade if you need autoscaling v2beta2.

**Comment by mikkeloscar** [comment]
The bug report was closed in GKE and users stating it works since v1.15.7: https://issuetracker.google.com/issues/135624588
